ccm_web/server/src/api/api.controller.ts

```diff
@@ -19,4 +22,24 @@ export class APIController {
   getGlobals (@Session() session: SessionData): Globals {
     return this.apiService.getGlobals(session)
   }
+
+  @Get('course/:id/name')
+  async getCourseName (
+    @Param('id', ParseIntPipe) courseId: number, @Session() session: SessionData
+  ): Promise<string> {
+    const { userLoginId } = session.data
+    const result = await this.apiService.getCourseName(userLoginId, courseId)
+    if (result === null) throw new InternalServerErrorException('Error occurred while communicating with Canvas')
+    return result
+  }
+
+  @Put('course/:id/name')
+  async putCourseName (
+    @Param('id', ParseIntPipe) courseId: number, @Body() courseNameDto: CourseNameDto, @Session() session: SessionData
+  ): Promise<string> {
+    const { userLoginId } = session.data
+    const result = await this.apiService.putCourseName(userLoginId, courseId, courseNameDto.newName)
+    if (result === null) throw new InternalServerErrorException('Error occurred while communicating with Canvas')
+    return result
```

Comment:
based on my comments [here](https://github.com/tl-its-umich-edu/canvas-course-manager-next/pull/121/files?file-filters%5B%5D=.ts#r652996449)

I feel you should send a response object something like this 
` res.status(result.statusCode).json(result.statusMessage)`

This way the FE will identify error state and normal state. Otherwise it will end up in 200 even there is an error

Reply:
Okay, I will try to refactor a bit, add more return information. NestJS supplies some abstraction so you don't need to Express pattern you suggested in these type of endpoints, but I'll see if it makes sense to override them.

Reply:
I decided to use the NestJS `HTTPException` approach. See latest commits, and https://docs.nestjs.com/exception-filters#throwing-standard-exceptions